akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -1234,13 +1242,13 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
     @tailrec
     private def invokeWithPromise(event: T, promise: Promise[Done]): Unit =
       currentState.get() match {
-        case Initialized =>
+        case i if i eq Initialized =>
```

Comment:
???

`Initialized` is still an case object, why would that be needed? 

(btw, `i if i` is a funny read)

Reply:
It's because the ADT top type now has a type parameter and so matching directly on it leads to:

```
[error]  found   : akka.stream.stage.ConcurrentAsyncCallbackState.Initialized.type
[error]  required: akka.stream.stage.ConcurrentAsyncCallbackState.State[T]
[error] Note: Any >: T (and akka.stream.stage.ConcurrentAsyncCallbackState.Initialized.type <: akka.stream.stage.ConcurrentAsyncCallbackState.State[Any]), but trait State is invariant in type E.
[error] You may wish to define E as -E instead. (SLS 4.5)
[error]         case Initialized =>
```

And marking the type param invariant opens up a whole other can of worms I don't care much to solve.

Reply:
Figured out how I could type it to make it work